src/us/kbase/workspace/database/provenance/Common.java

```diff
@@ -47,38 +52,53 @@ static <T> List<T> processSimpleList(final List<T> list, final String name) {
 			return Common.immutable(list);
 		}
 	}
-	
+
+	static String checkRpid(final String putativeRpid, final String name) {
+		final String rpid = requireNonNull(processString(putativeRpid), name);
```

Comment:
Take a look at `us.kbase.workspace.database.Util.checkString`

Reply:
There's also the `checkURL` methods beneath this in this file. What's the logic behind what goes in `Common` vs what goes in `Util`?

Reply:
Whether it's only used in provenance classes or not. I imagine some will get moved to other places (I actually would like to change Util to something else at some point as it's not a very descriptive name) at some point.

Reply:
Renamed this function `checkPid` and altered the functionality to be more similar to the `checkString` function in `Util`.

Reply:
Oh, I just meant that you could replace `requireNonNull(processString...)` with `checkString`

Reply:
:+1: